### ISSUE TITLE
[Suggestion] --auto-correct shouldn't fix `Style/SymbolProc`

### ISSUE THREAD

**Issue by Malabarba** [bbatsov/rubocop#5009]
This is similar to (but not the same as) #2720.

Passing a block that simply calls a method (`{ |a| a.method }`) is not the same as passing a proc generated from the method name (`&:inputs`). The former has arity `1`, the latter has arity `-1`.
I'm not arguing for the removal of this rule, but, given that it can actually break code, I think that it shouldn't be auto-corrected by default.

Here's an example of real code that breaks under this rule. 

This is a simple way to define a form in ActiveAdmin for a model that `belongs_to :address`
```ruby
form do |f|
  f.inputs('User Address', for: :address) { |a| a.inputs }
  f.actions
end
```

Meanwhile, this raises an error (nowadays it's at least an informative error message, but on previous versions of ActiveAdmin the error was quite a bit more elusive).
```ruby
form do |f|
  f.inputs('User Address', for: :address,  &:inputs)
  f.actions
end
```

--------

## Expected behavior

I would expect `bundle exec rubocop --auto-correct` to not make any changes that might break code.

## RuboCop version

```
0.50.0 (using Parser 2.4.0.0, running on ruby 2.2.5 x86_64-linux)
```

**Comment by Drenmi** [comment]
Thanks for the report!

This is happening because ActiveSupport, for unknown reasons, overrides behaviour based on block arity. We've had quite a [lengthy discussion about it](https://github.com/bbatsov/rubocop/issues/3071).

Because RuboCop is a Ruby analyzer, if we can avoid it we really do not want to hard code special cases for different frameworks, even in the configuration.

The solution right now is to add offending methods to the `IgnoredMethods` configuration option.

I am also working on a recommended configuration for Rails, which will deal with this cop and many other common false positives.

**Comment by Malabarba** [comment]
Thanks for the quick reply.


I know how lengthy these OSS discussions can get, and I have no intention of dragging you into another one. 

Thanks for pointing me towards a workaround, and thanks for Rubocop.


If you'd like to add my opinion to your pool of thoughts, I've written it below. But please don't feel obliged to reply at all. I respect you've probably put a lot more thought and time into this than me. 


---


I disagree that this is just a rails thing. Any user code could handle a block differently depending on its arity. At the end of the day, `proc {|x| x.method}` is simply not the same as `:method.to_proc`, and automatically replacing one with the other will always have the possibility of introducing bugs (rails or no rails). 

And thanks again!

**Comment by Drenmi** [comment]
Thanks for the feedback @Malabarba!

I will add your suggestion to disable auto-correct by default to my list of things to think about. 🙂 It is actually quite in line with what we are doing already. Erring on the side of caution with corrections, that is.
